metron-stellar/stellar-common/src/main/java/org/apache/metron/stellar/common/shell/cli/PausableInput.java

```diff
@@ -36,8 +37,8 @@
  *
  */
 public class PausableInput extends InputStream {
-  InputStream in = System.in;
-  boolean paused = false;
+  private InputStream in = System.in;
+  private AtomicBoolean paused = new AtomicBoolean(false);
```

Comment:
What problem were you solving here @ottobackwards ?  Is this bit access by multiple threads?

Reply:
I found in travis and locally, that the PausableInput was hanging, my builds where not completing.      PausableInput.INSTANCE.unpause(); was never returning.

I couldn't figure out why it was locked.  Reading the class documentation and the trying to understand the threading in the shell, I looked at the class and saw some inconsistencies with how and when we checked if we where paused in the different read() calls, and also that the flag field was not volatile or atomic.  I refactored this things to what I *think* would be the correct and consistent approaches and .... it resolved my issue.

Reply:
Good find!  I'm sure that was frustrating to dig into.